src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+    mMptzPosition.tilt = Optional<int16_t>(defaultTilt);
+    mMptzPosition.zoom = Optional<uint8_t>(defaultZoom);
+
+    VerifyOrReturnError(AttributeAccessInterfaceRegistry::Instance().Register(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::HasFeature(Feature aFeature) const
+{
+    return mFeatures.Has(aFeature);
+}
+
+bool CameraAvSettingsUserLevelMgmtServer::SupportsOptAttr(OptionalAttributes aOptionalAttr) const
+{
+    return mOptionalAttrs.Has(aOptionalAttr);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::MarkDirty(AttributeId aAttributeId)
+{
+    MatterReportingAttributeChangeCallback(mEndpointId, CameraAvSettingsUserLevelManagement::Id, aAttributeId);
+}
+
+// Attribute mutators
+//
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetMaxPresets(uint8_t aMaxPresets)
+{
+    mMaxPresets = aMaxPresets;
+    MarkDirty(Attributes::MaxPresets::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetTiltMin(int16_t aTiltMin)
+{
+    if ((aTiltMin < kMinTiltValue) || (aTiltMin > kMaxTiltValue - 1) || (aTiltMin > mTiltMax))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mTiltMin = aTiltMin;
+    MarkDirty(Attributes::TiltMin::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetTiltMax(int16_t aTiltMax)
+{
+    if ((aTiltMax > kMaxTiltValue) || (aTiltMax < kMinTiltValue + 1) || (aTiltMax < mTiltMin))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mTiltMax = aTiltMax;
+    MarkDirty(Attributes::TiltMax::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetPanMin(int16_t aPanMin)
+{
+    if ((aPanMin < kMinPanValue) || (aPanMin > kMaxPanValue - 1) || (aPanMin > mPanMax))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mPanMin = aPanMin;
+    MarkDirty(Attributes::PanMin::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetPanMax(int16_t aPanMax)
+{
+    if ((aPanMax > kMaxPanValue) || (aPanMax < kMinPanValue + 1) || (aPanMax < mPanMin))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mPanMax = aPanMax;
+    MarkDirty(Attributes::PanMax::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CameraAvSettingsUserLevelMgmtServer::SetZoomMax(uint8_t aZoomMax)
+{
+    if ((aZoomMax > kMaxZoomValue) || (aZoomMax < kMinZoomValue + 1))
+    {
+        return CHIP_IM_GLOBAL_STATUS(ConstraintError);
+    }
+
+    mZoomMax = aZoomMax;
+    MarkDirty(Attributes::ZoomMax::Id);
+    return CHIP_NO_ERROR;
+}
+
+// Mutators that may be invoked by a delegate in responding to command callbacks, or due to local on device changes
+//
+void CameraAvSettingsUserLevelMgmtServer::SetPan(Optional<int16_t> aPan)
+{
+    if (aPan.HasValue())
```

Comment:
So what if the value is missing?  Shouldn't we still set it in mMptzPosition?  I'm not sure why this takes an Optional at all, but then ignores missing values...

This applies to a number of attributes.

Maybe this needs some documentation in the header explaining the behavior and why that behavior is the way it is?  It sounds like the goal is to have the member never have a missing value?

Reply:
Added a comment above these methods.  The value can be missing, it presence depends on feature flags.  An attempt to write a missing value can happen,e.g. an MPTZSetPosition with only one of the three being set.  This is basically ignored.